### DISCUSSION TITLE
But the DOM *does* detect the line breaks...?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#8483] [General]
The perennial question about line breaks... but might this be a new discovery?

Using the inspector tool on a some transcluded "soft-linebroken" rows I see this:

<img width="379" alt="image" src="https://github.com/user-attachments/assets/e515f6cd-6d19-4b32-9362-fecdfa106f8d">

<br>...buuuut, if I click inside the inspector, on those words, it splits up at the very soft linebreaks:

<img width="353" alt="image" src="https://github.com/user-attachments/assets/4abde8f1-9ddc-4535-a3b6-1aa2a3c056ce">

...so the system (the DOM?) **does** know there are line breaks there!?

Could perhaps TW utilize this to "allow" soft linebreaks after all?

**Comment by pmario** [comment]
See: https://tiddlywiki.com/#Hard%20Linebreaks%20with%20CSS

`white-space: pre-wrap;` is the key rule here. https://developer.mozilla.org/en-US/docs/Web/CSS/white-space

**Comment by twMat** [comment]
Oh, thanks @pmario - somehow I had missed that tiddler and feature! (from 2017!)

It keeps popping up as a question, especially for noobs of course, so I was of the impression that it was not "solved". 

A tag for this would, in many cases, be visually cluttering but this just gave me a super idea to improve TW hackability for CSS:

Let's make it a *general* feature that **fields with prefix `data-` are added to the tiddler div** next to `data-tiddler-title` and `data-tags`!

That would enable users to make *custom "special fields"* to be used in css! (For example, `data-linebreaks:yes` could then be made to deal with the OP.)

As should be evident from my plugin development over the years, I have discovered how powerful CSS is in TW. JS is basically blocked, and it is not recommended to overwrite shadows, so CSS is the perfect powertool to basically harmlessly  customize stuff. 

What do you think about this?

**Reply by saqimtiaz** [reply]
> Let's make it a _general_ feature that **fields with prefix `data-` are added to the tiddler div** next to `data-tiddler-title` and `data-tags`!

The [ClassFilters](https://tiddlywiki.com/#SystemTag%3A%20%24%3A%2Ftags%2FClassFilters%2FTiddlerTemplate) mechanism was designed to allow exactly such use cases without the need to hardcode individual bespoke solutions that would increase the complexity of the core templates.

**Comment by twMat** [comment]
>The [ClassFilters](https://tiddlywiki.com/#SystemTag%3A%20%24%3A%2Ftags%2FClassFilters%2FTiddlerTemplate) mechanism was designed to allow exactly

Thanks @saqimtiaz ! OK, great! I attempted to try it out but didn't get anywhere. The brief description in that tiddler is way too cryptic for me to understand and there is no example provided there. (This comment is in no way a request for anyone to "fix" that at this time, and I don't have a use case. I might return to the matter at some later time.)

**Reply by pmario** [reply]
>This comment is in no way a request for anyone to "fix" that at this time, and I don't have a use case. I might return to the matter at some later time.

I can remember, when it was implemented. But reading the info now. I also thought, there should be an example. .. I had to look up the code to see how it would work.
